crates/subspace-farmer/src/bin/subspace-farmer/commands/shared/network.rs

```diff
@@ -163,6 +163,10 @@ where
                 async move {
                     let internal_result = match req {
                         SegmentHeaderRequest::SegmentIndexes { segment_indexes } => {
+                            if segment_indexes.len() as u64 > SEGMENT_HEADER_NUMBER_LIMIT {
+                                return None;
```

Comment:
Add the debug message here, please, similar to what we have in `SegmentHeaderRequest::LastSegmentHeaders`